src/hotspot/share/utilities/vmError.cpp

```diff
@@ -350,7 +350,7 @@ static frame next_frame(frame fr, Thread* t) {
 void VMError::print_native_stack(outputStream* st, frame fr, Thread* t, bool print_source_info, char* buf, int buf_size) {
 
   // see if it's a valid frame
-  if (fr.pc()) {
+  if (fr.sp() != nullptr && os::is_readable_pointer(fr.sp())) {
```

Comment:
Why did you completely change the notion of "valid frame" here?

Reply:
I needed for an early version of the patch, but I can remove it now.  Using sp() instead of pc() could possibly give more progress on weird native frames, and is_readable_pointer() prevents some crashes during stack walking.